spec/spec_helper.rb

```diff
@@ -57,6 +57,15 @@
   config.mock_with :rspec do |mocks|
     mocks.yield_receiver_to_any_instance_implementation_blocks = true
   end
+
+  # Monkey-patch Marshal.dump to enable dumping of mocked or stubbed objects. By
+  # default this will not work since RSpec mocks works by adding singleton methods
+  # that cannot be serialized. This patch removes these singleton methods before
+  # serialization. Potentially but unlikely to be removed post upgrade.
+  config.mock_with :rspec do |mocks|
+    mocks.patch_marshal_to_support_partial_doubles = true
```

Comment:
@chrisarcand is this still in the latest rspec version? 3.4?  I'm curious if we will need to disable this option at some point, fix any failures by not marshalling mock objects, and remove this option...